webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/JsonQueryObjects.scala

```diff
@@ -334,14 +371,62 @@ object JsonQueryObjects {
     }
   }
 
-  // policy servers are serialized in their output format
+  final case class JQNodeDetailLevel(
+      fields: Set[String]
+  ) extends AnyVal
+
+  object JQNodeDetailLevel {
+    implicit val transformer: Transformer[JQNodeDetailLevel, NodeDetailLevel] = (jq: JQNodeDetailLevel) => {
+      import jq.fields
+      if (fields.contains("full")) {
+        FullDetailLevel
+      } else {
+        val base         = {
+          if (fields.contains("minimal")) {
+            MinimalDetailLevel
+          } else {
+            DefaultDetailLevel
+          }
+        }
+        val customFields = fields.filter { field =>
+          field != "minimal" &&
```

Comment:
I'm not sure why you special-case `minimal` and `default` here. From the field's name point of view, they are invalid like "foo" could be, and you already took care of them aboce. 

It won't have any impact on perf either, since `allFields` is not big, the hash resolution will be quick. 
Is there any other reason for doing so I don't see ?

Reply:
This whole portion of code was copied from the previous [RestExtractorService method](https://github.com/Normation/rudder/blob/branches/rudder/8.1/webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/RestExtractorService.scala#L276) to "parse" the node details level...

Right below we actually need to separate the input set of string into either : 
* special values ("full", "minimal", "default")
* custom values ("bios", etc.), which we need to validate before passing them to `CustomDetailLevel(special, custom)`
=> Maybe what we want here is a smarter constructor for `CustomDetailLevel`, which is the actual structure that needs the splitting

Reply:
(and I forgot to remove the stale `RestExtractorService` method which is now unused)